Add HL7 aliases files

masnick opened the pull request.

cmoesel commented:
Hey @masnick.  This is cool!  I don't have time to review the scripts right now (also, I don't actually know Python) -- but a I noticed a few things looking at the generated files:
* They are missing the `// @Name:` and `// @Description:` conventions seen in the other example files
* They don't prefix aliases w/ `$`, but the I think we are now encouraging people to use `$` (see pre-existing alias examples)
* The alias names have `-` in them. I'm not sure this is bad, but it's something we haven't generally done (nor seen done).  I don't know if there is any better alternative, but I thought I'd bring it up for discussion.  @markkramerus, any thoughts?
* Some of the alias names are very long, but since they are generated, I'm assuming there is not much we can do except maybe hand modify the really long ones after... if that even makes sense.

I'm also wondering if the scripts really should live under the `Examples/Aliases` folder.  I think it would be better to keep everything under `Examples` as true examples and move scripts to a top-level `scripts` (or `Scripts`?) folder; e.g., `/Scripts/Aliases/*.*`.

Again, thanks for doing this.  It's very cool!  And maybe if I get a chance to review the scripts later, I'll learn some Python!

masnick commented:
Thanks, @cmoesel!

I used Python for this because I thought having a Jupyter Notebook might be easier for people to run in the future vs. a script via the command line. However, it turns out the dependency management still requires the command line, so I don't know how useful this is after all 🙃 

In any case, as long as you're happy with the output, I don't think the code needs an extensive review. The logic is pretty straight-forward, and could easily be converted to Ruby or something else.

I've made the following changes:

- Added the `// @Name` and `// @Description` frontmatter
- Added `$` to the front of the alias names
- Moved the notebook into a separate `/Scripts/Aliases/` folder, and renamed to `HL7.ipynb`. This mirrors the `/Examples/Aliases/` structure and the `HL7-*.fsh` filenames.

For your comments on the alias names:

- I agree that the super long alias names are not ideal. We could easily add manual remappings if someone wants to create them.
- The dashes come from two sources: the slugs defined in the HL7 terminology, and the `v2-`/`v3-` prefixes. I think `v2-someSlug` is more readable than `v2someSlug`, so if we're allowing dashes at all I think it's a good idea to use them for the prefix. I'm reluctant to start modifying the slugs to remove dashes because it will be a bit fiddly to get the capitalization right so they are still readable, but this is probably doable if we want to avoid dashes entirely.

markkramerus commented:
Approved.
